acpi.adoc

```diff
@@ -1,26 +1,142 @@
+
+This section defines mandatory and optional ACPI requirements on top of the ACPI specification cite:[ACPI]. Additional non-normative guidance may be found in <<ACPI Implementation Guidance>>.
+
+=== ACPI High-Level Requirements
+
+A compliant platform must:
+
+* Conform to UEFI specification requirements for ACPI:
+** RISC-V Platforms (cite:[UEFI] Section 2.3.7)
+** Industry Standard Configuration Tables (cite:[UEFI] Section 4.6.1.1)
+* Conform to version 6.6 of the ACPI specification, including RISC-V requirements listed in:
+** Multiple APIC Description Table (MADT, cite:[ACPI] Section 5.2.12)
+** RISC-V Hart Capabilities Table (RHCT, cite:[ACPI] Section 5.2.36)
+* Meet the requirements in <<Minimum Hardware Requirements Imposed By ACPI>>
+* Be 64-bits clean.
+** RSDT must not be implemented.
+** 32-bit address fields must be 0.
+* Implement a firmware-first APEI model (cite:[ACPI] Section 18.4)
+
+=== ACPI-specific Hardware Requirements
+
+This section lists the BRS-specific hardware requirements imposed by
+ACPI support.
+
+==== HW-Reduced ACPI Model
+
+Compliant RISC-V platforms only implement the HW-Reduced ACPI Model. This means the hardware portion of
+cite:[ACPI] (Section 4) is not required or supported. All functionality is instead provided through equivalent
+software-defined interfaces and the complexity in supporting ACPI is reduced.
+
+The following additional BRS requirements apply:
+
+* Both interrupt-signaled and GPIO-signaled ACPI Events are supported
+  (cite:[ACPI] Section 5.6)
+* There is at least 1 wake event.
+** Note: for systems that do not support Sx states except S5 soft off, this can be just the power button.
+* For the ACPI Platform Error Interfaces (APEI) support:
+** There is 1 event for non-fatal error signaling (cite:[ACPI] Section 18.3.2.7.2)
+** There is 1 NMI-equivalent signal for use in fatal errors (cite:[ACPI] Section 18)
+
+==== Hart Performance Control
+
+If the platform provides support for OS-directed hart performance control, then it must be exposed using
+Collaborative Processor Performance Control (CPPC, cite:[ACPI] Section 8).
+
+==== Time and Alarm Device
+
+UEFI Runtime Service should be used to provide time services to the
+OS, unless the RTC is subject to arbitration issues between concurrent
+OS and UEFI accesses to the underlying hardware. See <<<UEFI Real-Time
+Clock>>>.
+
+In situations where the TAD device depends on a vendor-specific OS
+driver for correct function (SPI, I2C, etc), the TAD must be
+function if the OS driver is not loaded. E.g. when a dependent
+driver is loaded, an AML method switches further accesses to go
+through the driver-backed OperationRegion.
+
+=== Additional ACPI Table Requirements
+
+This section lists additional BRS-specific requirements
+for the mandatory and conditionally-required ACPI tables in a compliant
+platform.
```

Comment:
Maybe stay away from 'platform' and use 'system'?

Reply:
Ok. Let's consistently use something. 'System', 'solution' or 'implementation'?

Reply:
I was using 'system', but 'implementation' may be better.

Reply:
wdyt think w.r.t. system vs implementation?

Reply:
So I'll leave this out as a separate change since IIUC we're still thinking of what the term should be,

Reply:
Feel free to pick one and run w/ it.

Reply:
Okay I went with 'system'.